### PR TITLE
Fix offline leaflet

### LINKED ISSUE #2419: Leaflet issues in offline mode
### Have you checked that your issue isn't already filed?

- [X] I read through [FAQ](https://github.com/alshedivat/al-folio/blob/master/FAQ.md) and searched through the [past issues](https://github.com/alshedivat/al-folio/issues), none of which addressed my issue.
- [X] Yes, I have checked that this issue isn't already filed.

### Bug description

Offline mode results in broken integrity hashes for Leaflet lib includes and graphics are missing. 

### How to reproduce the bug

1. Enable offline mode in _config.yml:
    ```
    third_party_libraries:
      download: true
    ```
2. Build static page
3. Open some page with Leaflet map
-> integrity errors
4. Remove integrity hashes manually
-> missing image (marker) files

### Error messages and logs

```
Failed to find a valid digest in the 'integrity' attribute for resource '.../assets/libs/leaflet.css' with computed SHA-256 integrity 'yuviegZOf3HiZ8eBDv629tOtj/DL9pjhq9m/8qjYX4o='. The resource has been blocked.
teaching/:1 Failed to find a valid digest in the 'integrity' attribute for resource '.../assets/libs/leaflet.js' with computed SHA-256 integrity 'fh+o7yCaRelV5DCgj6OhsmjH6qYlpiwY101vi/Ijeew='. The resource has been blocked.
teaching/:144 Uncaught ReferenceError: L is not defined
    at teaching/:144:2453
    at NodeList.forEach (<anonymous>)
    at document.onreadystatechange (teaching/:144:2297)
```

It looks as if the files have actually been changed, at least in the first line.

After deactivating the check, some further problem with missing images (icons/markers) comes up:
I think just those from e.g. https://cdn.jsdelivr.net/npm/leaflet@1.9.4/dist/images/ need to be downloaded to /assets/libs/. Manually copying fixed this in my case.

### What operating system are you using?

Not applicable (e.g. you're using GitHub Pages or other hosting)

### Where are you seeing the problem on?

Deployed site

### More info

Started request in #1181 and was asked to open new issue.

### PR DESCRIPTION
**Pull request by george-gca**
Fixes #2419 by downloading leaflet images. Also changed the path where the libraries are downloaded to avoid not downloading files with same name.

### CODE REVIEW

File: _includes/scripts/tikzjax.liquid [hunk 1]
```diff
@@ -1,3 +1,8 @@
 {% if page.tikzjax %}
-  <script defer src="https://tikzjax.com/v1/tikzjax.js"></script>
+  <script
+    defer
+    src="https://tikzjax.com/v1/tikzjax.js"
```
Comment (CheariX) [inline _includes/scripts/tikzjax.liquid:4]: I'm wondering if this would not break offline mode?
The `src` seems to be hardcoded.

**Comment by CheariX** [comment]
I'm wondering how the images can downloaded beforehand?
I assume that only the initial images are downloaded but when you scroll the map, there will be missing images?

File: _includes/scripts/tikzjax.liquid (hunk 1 above)
Comment (george-gca) [reply _includes/scripts/tikzjax.liquid:4]: There's nothing much we can do about tikzjax. Converting it to be 100% offline would involve too much change to its original source code.

**Comment by george-gca** [comment]
Yes, there is so much we can do about offline support.

**Pull request merged**